gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+                     "id": "3b0ef8f4-82c7-44d4-a4fb-6177f9a21977",
+                     "name": "",
+                     "status": "PENDING_CREATE",
+                     "router_ids": [
+                         "650bfd2f-7766-4a0d-839f-218f33e16998"
+                     ],
+                     "tenant_id": "45977fa2dbd7482098dd68d0d8970117",
+                     "firewall_rule_list": True,
+                     "description": '{\
+                                    "vm_management_ip": "172.24.4.5",\
+                                    "service_vendor": "vyos"}',
+                     "firewall_rule_list": True
+                    }
+        return firewall
+
+    def fake_request_data_generic_config(self):
+        data = {
+                'context': {},
+                'kwargs': self._fake_kwargs(),
+                'request_info': 'xxx'}
+
+        request_data = {
+                 'info':   {
+                             'version': 'v1'},
+
+                 'config': [
+                        {'resource': 'interfaces',
+                         'kwargs': data}
+                    ]
+                }
+
+        return request_data
+
+
+class ConfiguratorRpcManagerTestCase(unittest.TestCase):
+    ''' Configurator RPC receiver '''
+
+    def __init__(self, *args, **kwargs):
+        super(ConfiguratorRpcManagerTestCase, self).__init__(*args, **kwargs)
+        self.fo = FakeObjects()
+
+    @mock.patch(__name__ + '.FakeObjects.conf')
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    def _get_ConfiguratorRpcManager_object(self,sc, conf):
+        cm = cfgr.ConfiguratorModule(sc)
+        demuxer = demuxer_lib.ConfiguratorDemuxer()
+        rpc_mgr = cfgr.ConfiguratorRpcManager(sc, cm, conf, demuxer)
+        return sc, conf, rpc_mgr
+
+    def _get_GenericConfigRpcManager_object(self, conf, sc):
+        agent = gc.GenericConfigRpcManager(sc, conf)
+        return agent, sc
+
+    @mock.patch(__name__ + '.FakeObjects.nqueue')
+    @mock.patch(__name__ + '.FakeObjects.drivers')
+    def _get_GenericConfigEventHandler_object(self, sc, rpcmgr, drivers, nqueue):
+        agent = gc.GenericConfigEventHandler(sc, drivers, rpcmgr, nqueue)
+        return agent
+
+    def _get_FWaasRpcManager_object(self, conf, sc):
+        agent = fw.FWaasRpcManager(sc, conf)
+        return agent, sc
+
+    def _test_generic_event_creation(self, operation, method, bulk=False):
+        sc, conf, rpc_mgr = self._get_ConfiguratorRpcManager_object()
+        agent, sc = self._get_GenericConfigRpcManager_object(conf, sc)
+        
+        if bulk:
+            request_data = self.fo.fake_request_data_generic_bulk()
+            request_data1 = self.fo.fake_request_data_generic_bulk()
+        else:
+            request_data = self.fo.fake_request_data_generic_single()
+            request_data1 = self.fo.fake_request_data_generic_single()
+
+        if (method == 'CONFIGURE_ROUTES' or method == 'CLEAR_ROUTES'):
```

Comment:
It is good to keep these in a map. If we pass the method to the map, you will get the resource name. You can avoid conditions with this.